test/sql/pragma/test_no_wal_mode.test

```diff
@@ -0,0 +1,181 @@
+# name: test/sql/pragma/test_no_wal_mode.test
+# description: Measure WAL metrics, turn no_wal_mode on, check that WAL replay is not performed.
+# group: [pragma]
+
+require json
+
+require noforcestorage
+
+require skip_reload
+
+statement ok
+SET threads = 1;
+
+statement ok
+SET wal_autocheckpoint = '1TB';
+
+statement ok
+PRAGMA disable_checkpoint_on_shutdown;
+
+statement ok
+ATTACH '__TEST_DIR__/wal_latency.db' AS wal_latency;
+
+statement ok
+CREATE TABLE wal_latency.tbl AS SELECT range AS id, 0 AS v FROM range(500_000);
+
+# Perform a large change in a single transaction so COMMIT must write a big WAL
+statement ok
+BEGIN TRANSACTION;
+
+statement ok
+UPDATE wal_latency.tbl SET v = id;
+
+statement ok
+PRAGMA profiling_output = '__TEST_DIR__/wal_commit.json';
+
+statement ok
+PRAGMA custom_profiling_settings='{"COMMIT_WRITE_WAL_LATENCY": "true"}';
+
+statement ok
+SET profiling_coverage='ALL';
+
+statement ok
+PRAGMA enable_profiling = 'json';
+
+statement ok
+COMMIT;
+
+statement ok
+PRAGMA disable_profiling;
+
+statement ok
+CREATE OR REPLACE TABLE commit_metrics AS SELECT * FROM '__TEST_DIR__/wal_commit.json';
+
+query I
+SELECT CASE WHEN commit_write_wal_latency > 0 THEN 'true' ELSE 'false' END FROM commit_metrics;
```

Comment:
I think this test can be simplified by, instead of using metrics, just verifying there is no WAL (i.e. `SELECT COUNT(*) FROM glob('__TEST_DIR__/wal_latency.db.wal) == 0`)